### PR TITLE
Remove requirement for password on MySQL

### PR DESCRIPTION
**Pull request by Mrod1598**
**Description:** <Describe what has changed.>
Removed requirement for password and set defaults for username and endpoint.

**Testing**
To test manually,[ install mysql server](https://www.digitalocean.com/community/tutorials/how-to-install-the-latest-mysql-on-debian-10) and when prompted for a password, don't set one. this will default to using the unix socket. The default location on debian is `/var/run/mysqld/mysqld.sock`. configure the the collector to collect from that socket using `root` as the username.

### CODE REVIEW

File: receiver/mysqlreceiver/README.md [hunk 1]
```diff
@@ -14,12 +14,11 @@ Collecting most metrics requires the ability to execute `SHOW GLOBAL STATUS`. Th
...
 
-The following settings are required to create a database connection:
-- `endpoint`
-- `username`
-- `password`
 
 The following settings are optional:
+- `endpoint`: (default = `localhost:3306`)
+- `username`: (default = `root`)
+- `password`: The password to the username. If not specified, root/admin access is required.
```
Comment (djaglowski) [inline receiver/mysqlreceiver/README.md:21]: > If not specified, root/admin access is required.

Is this necessarily true? Obviously not recommended, but could you have a non-root user w/o a password?

File: receiver/mysqlreceiver/README.md [hunk 2]
```diff
@@ -38,6 +37,13 @@ receivers:
     collection_interval: 10s
 ```
 
+### Example Configuration With Least Configuration
+
+```yaml
+receivers:
+  mysql:
+```
+
```
Comment (djaglowski) [inline receiver/mysqlreceiver/README.md:46]: I would omit this.

File: receiver/mysqlreceiver/README.md (hunk 1 above)
Comment (Mrod1598) [reply receiver/mysqlreceiver/README.md:21]: actually yes, found out if you set authentication to `auth_socket` you can just use the unix socket endpoint. I'll update it.

File: receiver/mysqlreceiver/README.md (hunk 2 above)
Comment (Mrod1598) [reply receiver/mysqlreceiver/README.md:46]: removed

**Review by djaglowski — APPROVED** [review]
Thanks @Mrod1598. LGTM

**Review by dehaansa — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**